content/rosa/entra-external-auth/index.md

```diff
@@ -0,0 +1,455 @@
+
+```bash
+GROUP_ID=<value from above>
+```
+
+In the Azure Portal, search for Enterprise Applications and then the external auth application you previously created.
+
+![Azure Portal - All Applications](./images/azure-portal-all-applications.png)
+
+Click on Assign users and groups
+![Azure Portal - assign users](./images/azure-portal-assign-groups.png)
+
+Click on Add user/group
+![Azure Portal - Add Group](./images/azure-portal-add-group.png)
+
+Select the group you want to add to openshift
+![Azure Portal - Select Group](./images/azure-portal-select-group.png)
+
+
+Next, we will create a `ClusterRoleBinding` that grants the members of the above group the ability to use the `cluster-admin` `ClusterRole`.
+
+```bash
+oc apply -f - <<EOF
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: rosa-admins
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+- apiGroup: rbac.authorization.k8s.io
+  kind: Group
+  name: $GROUP_ID
+EOF
+```
+
+### Validate External Authentication
+
+#### Validate using the OpenShift Web Console
+
+To validate you can log into the OpenShift console with Microsoft Entra ID, first retrieve the OpenShift Console URL by running the following command:
+
+```bash
+rosa describe cluster -c ${ROSA_CLUSTER_NAME} -o json | jq -r '.console.url'
+```
+
+Open this URL and your external authentication provider should open.
+
+### Validate access to the cluster's API
+Here we will be leveraging a client-go Credential Plugin to automate the interaction between the OpenShift CLI and the External Authentication Provider including the management of OIDC tokens.
+
+More information related to Kubernetes plugins can be found [here](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/).
+
+Once the plugin has been configured and is available, create a KUBECONFIG file here with EntraID details for example create **rosa-auth.kubeconfig** file with following information
+
+```bash
+apiVersion: v1
+clusters:
+- cluster:
+    server: ${API_URL}
+  name: cluster
+contexts:
+- context:
+    cluster: cluster
+    namespace: default
+    user: oidc
+  name: admin
+current-context: admin
+kind: Config
+preferences: {}
+users:
+- name: oidc
+  user:
+    exec:
+      apiVersion: client.authentication.k8s.io/v1
+      args:
+      - oidc-login
+      - get-token
+      - --oidc-issuer-url=https://login.microsoftonline.com/${TENANT_ID}/v2.0
+      - --oidc-client-id=${CLIENT_ID}
+      - --oidc-client-secret=${CLIENT_SECRET}
+      - --oidc-extra-scope=email
+      - --oidc-extra-scope=openid
+      command: kubectl
+      env: null
+      interactiveMode: Never
+```
+Set the KUBECONFIG environment variable to the location of the rosa-cluster.kubeconfig file which will enable the OpenShift CLI to authenticate against the ROSA cluster with the contents of the file.
```

Comment:
```suggestion
Set the `KUBECONFIG` environment variable to the location of the `rosa-cluster.kubeconfig` file. This will configure the OpenShift CLI to authenticate against the ROSA cluster with the OIDC client.
```